src/interfaces/init.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_INTERFACES_INIT_H
+#define BITCOIN_INTERFACES_INIT_H
+
+#include <fs.h>
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+struct NodeContext;
+
+namespace interfaces {
+class Base;
+class IpcProcess;
+class IpcProtocol;
+
+//! Init interface providing access to other interfaces. The Init interface is
+//! the interface first exposed when a process spawns or connects to another
+//! process.
+//!
+//! When spawning a new process, the steps are:
+//!
+//! 1. Client process calls IpcProcess::spawn(), which spawns a new process and
+//!    returns a socketpair file descriptor for communicating with it.
+//! 2. Client process calls IpcProtocol::connect() passing the socketpair
+//!    descriptor, which returns a client Init proxy object calling remote Init
+//!    interface methods.
+//! 3. Client process calls client Init proxy object methods to make new client
+//!    proxy objects calling other remote interfaces. It can also destroy the
+//!    Init object to shut down the spawned process.
+//! 4. Spawned process calls IpcProcess::serve(), to read command line arguments
+//!    and determine whether it is a spawned process and what socketpair file
+//!    descriptor it should use. It calls IpcProtocol::serve() to handle
+//!    incoming requests from the socketpair and call Init interface methods, or
```

Comment:
It's unclear who is the "it" here calling Init interface methods, still the spawned subprocess ? Or do you mean the spawned subprocess has the option to shutdown itself ?